### PR TITLE
Various doc fixes enhancing IDE experience, fixed wrong class name

### PR DESCRIPTION
**Pull request by grisha87**
- Added PHPDoc typehints powering up IDE auto-completion
- Fixed reference to undefined class `EqualsTo` - replaced it with `EqualTo`
- Unified `bool` / `int` typehints
- Fixed spotted typo

### CODE REVIEW

File: lib/Qandidate/Toggle/Serializer.php [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+namespace Qandidate\Toggle;
+
+/**
+ * Representation of a serializer
+ *
+ * This class provides common ground for concrete serializer implementations
+ *
+ * @TODO: serialize() and deserialize() should be part of abstract interface
+ */
+abstract class Serializer
```
Comment (asm89) [inline lib/Qandidate/Toggle/Serializer.php]: This class is currently only introduces to re-use `assertHasKey()`. I'd rather move the `assertHasKey()` method to a utility function / static function in a helper class somewhere. The `@todo` mentions making `serialize()` and `deserialize()` part of the abstract interface, but the signature will be different per implementation, so that doesn't make sense?

Unless there are other arguments for introducing this base class I'd prefer to not introduce it.

File: lib/Qandidate/Toggle/Operator/EqualTo.php [hunk 2]
```diff
@@ -1,5 +1,14 @@
 <?php
 
+/*
+ * This file is part of the qandidate/toggle package.
+ *
+ * (c) Qandidate.com <opensource@qandidate.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
```
Comment (asm89) [inline lib/Qandidate/Toggle/Operator/EqualTo.php]: Good catch. Thank you!

File: lib/Qandidate/Toggle/Serializer/OperatorSerializer.php [hunk 3]
```diff
@@ -35,7 +36,7 @@ class OperatorSerializer
     public function serialize(Operator $operator)
     {
         switch(true) {
-            case $operator instanceof EqualsTo:
+            case $operator instanceof EqualTo:
                 return array('name' => 'equals-to', 'value' => $operator->getValue());
```
Comment (asm89) [inline lib/Qandidate/Toggle/Serializer/OperatorSerializer.php]: Let's fix it here as well. E.g. `equals-to` => `equal-to`.

File: lib/Qandidate/Toggle/Serializer/OperatorSerializer.php [hunk 4]
```diff
@@ -69,7 +70,7 @@ public function deserialize(array $operator)
             case 'equals-to':
```
Comment (asm89) [inline lib/Qandidate/Toggle/Serializer/OperatorSerializer.php]: Given we "fix" `equals-to` above. Let's add the `equal-to` case here, but also keep the old one for backwards compatibility.

File: lib/Qandidate/Toggle/Serializer.php (hunk 1 above)
Comment (grisha87) [inline lib/Qandidate/Toggle/Serializer.php]: Yep. This one got into the PR since I wasn't using a branch for the PR itself. `AssertionHelper` then. 

When we're talking about signatures - these would have to be unified, and argument validation should be carried out by child classes. I don't think it's the time for this change yet - unless I'll need to do some generic mocks.

Comment (asm89) [inline lib/Qandidate/Toggle/Serializer.php]: If you have a concrete proposal for that can you put it in a separate PR? I'm happy to merge the phpdoc fixes + the `EqualsTo` bug fix, but I'd like to further discuss the new abstract class. :)

**Comment by grisha87** [comment]
I've reverted adding the new class. Further modifications / clean-ups will come as separate PRs (if they will be useful / required).

**Comment by asm89** [comment]
@grisha87 Can you squash your commits?

**Comment by grisha87** [comment]
Here yo go mate :)

**Comment by othillo** [comment]
+1
